datacats/error.py

```diff
@@ -4,7 +4,10 @@
 class DatacatsError(Exception):
 
     def __init__(self, message, format_args=(), parent_exception=None):
-        self.message = message
+        try:
+            self.message = message.format(*format_args)
+        except (ValueError, KeyError):
```

Comment:
Both those cases would be our code with an error that shouldn't be silently ignored.

Better yet: let's remove format_args from the constructor and let the code that creates these instances call .format() if it wants to on the message it passes in. That way the error stays right where it was caused and unrelated code stays out of our exception classes.